### DISCUSSION TITLE
Using Otel 1.26.0, getting `Failed to connect to localhost/127.0.0.1:4317` error

### DISCUSSION THREAD

**Discussion by itsmaneesh** [open-telemetry/opentelemetry-java-instrumentation#8708] [Q&A] [answered]
I have been using `Opentelemetry-javaagent:1.10.0` with the following env variables and auto-instrumentation was working fine and no error in the app logs or issue in exporting the `spans`. (Note, I have NOT configured the env var for metrics exporter i.e.: `OTEL_METRICS_EXPORTER`)

```
JAVA_TOOL_OPTIONS:                                                          -javaagent:/app/libs/opentelemetry-javaagent-1.10.0.jar
OTEL_EXPORTER_JAEGER_ENDPOINT:                                              http://jaeger.cluster.local:14250
OTEL_TRACES_EXPORTER:                                                       jaeger
```

But, I was trying to use Opentelemetry-java-agent:1.26.0 with the following env variables and auto-instrumentation works fine however there is continuous error in app logs in exporting `metrics`. (Note, I have NOT configured the env var for metrics exporter i.e.: `OTEL_METRICS_EXPORTER`)

```
JAVA_TOOL_OPTIONS:                                                          -javaagent:/app/libs/opentelemetry-javaagent-1.26.0.jar
OTEL_EXPORTER_JAEGER_ENDPOINT:                                              http://jaeger.cluster.local:14250
OTEL_TRACES_EXPORTER:                                                       jaeger
```

Error in app logs:

`[otel.javaagent 2023-06-13 18:34:27:440 +0000] [OkHttp http://localhost:4317/...] ERROR io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter - Failed to export metrics. The request could not be executed. Full error message: Failed to connect to localhost/127.0.0.1:4317 `

Question is: 

1. Why do I see the error in case of Opentelemetry-javaagent:1.26.0 and not in with Opentelemetry-javaagent:1.10.0 ? Was this due to the env var `OTEL_METRICS_EXPORTER` defaulted to `none` earlier in version `1.10.0`  of javaagent and now its defaulted `otlp` (change introduced in `1.13.0`)? If so, what was the rationale to do so?

Ref: https://github.com/open-telemetry/opentelemetry-java/pull/4355/files#diff-77389189e1151331dcf90859f30c3ffc8f9fef10b73c600e84db953446e878ff

2. If my metrics export is `prometheus` then can I configure the below env var to resolve this issue?
 - OTEL_METRICS_EXPORTER=prometheus
 - OTEL_EXPORTER_PROMETHEUS_PORT=<prometheus_port>
 - OTEL_EXPORTER_PROMETHEUS_HOST=<prometheus_host_name>

Thanks

**Comment by trask** [comment, marked as answer]
> 1. Why do I see the error in case of Opentelemetry-javaagent:1.26.0 and not in with Opentelemetry-javaagent:1.10.0 ? Was this due to the env var `OTEL_METRICS_EXPORTER` defaulted to `none` earlier in version `1.10.0`  of javaagent and now its defaulted `otlp` (change introduced in `1.13.0`)? If so, what was the rationale to do so?

yes

> If so, what was the rationale to do so?

metrics were enabled by default once they became stable

> 2. If my metrics export is `prometheus` then can I configure the below env var to resolve this issue?

check out https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#prometheus-exporter
